src/main/webapp/dev/js/main/feedbackSubmissionsEdit.js

```diff
@@ -518,32 +516,40 @@ function updateConstSumMessageQn(qnNum) {
     let allSame = true;
     let allNotNumbers = true;
     let answerSet = {};
+    const repeatedPoints = [];
 
     function fillWithZeroIfEmpty(inputFieldElement) {
         if (Number.isNaN(parseInt(inputFieldElement.val(), 10))) {
             inputFieldElement.val(0);
         }
     }
 
-    function checkAndDisplayMessage(messageElement) {
+    function checkAndDisplayMessage(instructionElement, messageElement) {
+        let instruction = '';
         let message = '';
+        const infoIcon = '<span class="glyphicon glyphicon-info-sign padding-right-10px"></span>';
+        const approvedIcon = '<span class="glyphicon glyphicon-ok padding-right-10px"></span>';
+        const errorIcon = '<span class="glyphicon glyphicon-remove padding-right-10px"></span>';
+
+        instruction += `${infoIcon} Total points distributed should add up to ${points}.<br>`;
+        if ((forceUnevenDistribution && !forceSomeUnevenDistribution) || forceAllUnevenDistribution) {
+            instruction += `${infoIcon} Every ${distributeToRecipients ? 'recipient' : 'option'} `
+                        + 'should be allocated different number of points.<br>';
+        }
+        if (forceSomeUnevenDistribution) {
+            instruction += `${infoIcon} At least one ${distributeToRecipients ? 'recipient' : 'option'} `
+                        + 'should be allocated different number of points.<br>';
+        }
+        instructionElement.addClass('text-color-blue');
+        instructionElement.removeClass('text-color-red');
+        instructionElement.removeClass('text-color-green');
 
-        if (allNotNumbers) {
-            message = `Please distribute ${points} points among the above `
-                    + `${distributeToRecipients ? 'recipients' : 'options'}.`;
-            messageElement.addClass('text-color-blue');
-            messageElement.removeClass('text-color-red');
-            messageElement.removeClass('text-color-green');
-        } else if (remainingPoints === 0) {
-            if (allowEvenDistribution || !forceUnevenDistribution
-                    || (forceUnevenDistribution && !forceSomeUnevenDistribution && allUnique)
-                    || (forceAllUnevenDistribution && allUnique)
-                    || (forceSomeUnevenDistribution && !allSame)) {
```

Comment:
`allowEvenDistribution` means some or all options can have the same points
`forceUnevenDistribution` means there have to be at least 2 options with different points
`forceAllUnevenDistribution` means everything has to be different
`forceSomeUnevenDistribution` is the intersection of `allowEvenDistribution`  and `forceUnevenDistribution`. 
As u can see, `forceSomeUnevenDistribution` can be true. So  `allowEvenDistribution`  and `forceUnevenDistribution` can be true at the same time.

I think you shld learn from this code fragment as you can tell each line i.e `allowEvenDistribution || !forceUnevenDistribution`, 
`(forceUnevenDistribution && !forceSomeUnevenDistribution && allUnique)`, `(forceAllUnevenDistribution && allUnique)` and `(forceSomeUnevenDistribution && !allSame)` each of them is a case of user input. Writing code is way is neater and easier for other contributors to read.

Reply:
Thanks for such a detailed explanation. Really helpful. I was performing a few experiments to get my head around this. I had some questions if that's alright.

To recap my understanding (please correct me if I am wrong):
- `forceSomeUnevenDistribution`: Implies there has to be a mixture, all can't be the same and all can't be different as well
- In `allowEvenDistribution`: at least two options have to be the same ? i.e. it is exclusive with`forceAllUnevenDistribution`
- When `forceSomeUnevenDistribution` is true both `allowEvenDistribution` and `forceUnevenDistribution` have to be true since it is the intersection

So first of all lets consider the `among recipients` question as given in one of the screenshots (question no. 6). When I created this question I selected the option `Every option to receive a different number of points`. Based on my understanding that would imply `forceAllUnevenDistribution`. I tried to print all these variables to the console for question 6 shown below:

`Question:  6`
`Allow even distribution:  true`
`Force uneven distribution:  true`
`Force some uneven distribution:  false`
`Force all uneven distribution:  false`

`forceAllUnevenDistribution` is not true against what I thought and same goes for `forceSomeUnevenDistribution`. Am i missing something here ?

Finally from the code fragment above aren't the conditions `(forceUnevenDistribution && !forceSomeUnevenDistribution && allUnique)` and `(forceAllUnevenDistribution && allUnique)` the same thing ?

Reply:
Similarly for question 7 I selected `At least some options to receive a different number of points`. I thought that basically means all can't be the same which implies `forceAllUnevenDistribution`, `forceSomeUnevenDistribution` and `forceUnevenDistribution` should be true. Only `allowEvenDistribution` should be false.

Reply:
hello @mg14777 , so sorry for the super late reply. I was quite caught up with my course. 

For the concepts, sorry i did not give a clearer explanation which led to your confusion. Thank you for clarifying with me. I think I gave some very confusing answer.

Now let me answer this again. I think somehow you have already figured some out. 

>forceSomeUnevenDistribution: Implies there has to be a mixture, all can't be the same and all can't be different as well
In allowEvenDistribution: at least two options have to be the same ? i.e. it is exclusive withforceAllUnevenDistribution
When forceSomeUnevenDistribution is true both allowEvenDistribution and forceUnevenDistribution have to be true since it is the intersection

`forceSomeUnevenDistribution` in the `Const.js` file, the explanation is 'At least some options'. This means all can't be the same, but all can be different.
`allowEvenDistribution` corresponds to not ticking the checkbox of "every option/at least some options to receive a different number of points", which means, the student can fill up in whatever way they want, without any restriction besides hving to reach the total points,

Basically, `forceSomeUnevenDistribution` ,  `allowEvenDistribution`  and `forceAllUnevenDistribution` are the three scenarios. You can say they are mutually exclusive in the sense that only one out of the three can be true, as you can tell that these boolean values are gotten by seeing if they match the string values.

```java
const allowEvenDistribution = constSumDistributePointsOptionSelected
            === ParamsNames.FEEDBACK_QUESTION_CONSTSUMNOUNEVENDISTRIBUTION;
    const forceAllUnevenDistribution = constSumDistributePointsOptionSelected
            === ParamsNames.FEEDBACK_QUESTION_CONSTSUMALLUNEVENDISTRIBUTION;
    const forceSomeUnevenDistribution = constSumDistributePointsOptionSelected
            === ParamsNames.FEEDBACK_QUESTION_CONSTSUMSOMEUNEVENDISTRIBUTION;
```

If you open the distribution questions from the student side, or `preview` as student, when you inspect the distribution questions, you will see a lot of hidden inputs inside each question. 
![image](https://user-images.githubusercontent.com/28526038/40306253-8be14b0c-5d30-11e8-8e99-b27a5853977a.png)

These hidden inputs are the indicators/flags for this question regarding its setting. You can find the use of these flags in method `updateConstSumMessageQn(qnNum)` in `feedbackSubmissionsEdit.js`.

In summary (sorry i m being so long winded), 

When `forceSomeUnevenDistribution` is true,  `allowEvenDistribution` is false, and `forceAllUnevenDistribution` is false,  and `forceUnevenDistribution` is false. 

When `allowEvenDistribution` is true, `forceSomeUnevenDistribution` and `forceAllUnevenDistribution` is false, and `forceUnevenDistribution` is false.

When `forceAllUnevenDistribution` is true, the other two cases are false and `forceUnevenDistribution` is false.

Reply:
as for this
>Finally from the code fragment above aren't the conditions (forceUnevenDistribution && !forceSomeUnevenDistribution && allUnique) and (forceAllUnevenDistribution && allUnique) the same thing ?

yeah they are the same thing. Firstly I suggest you not to use `forceUnevenDistribution`, as it is equivalent to `forceAllUnevenDistribution`. I think it's not cleared in the last PR regarding this. Our bad. But dont clear it in this PR, as this is not the concern of your issue. 

My concern is just out of readability. When there are too many booleans chained especially with negation, it's a bit more difficult to understand and get the logic right for those who read the code. We should aim to keep our code as simple as possible:))))

Reply:
@wenmogu Thanks a lot for clarifying my doubts. It's alright. I can imagine you must be busy. Apologies from my side as well for not being active for the past two days. I was partly awaiting clarification and partly preparing for the interview. But it's done now so it should be fine.

So in a nutshell, based on our discussion so far, I will try to keep the boolean expressions as simple as possible. I think there is still some confusion regarding mutual exclusiveness of the variables.
> Firstly I suggest you not to use forceUnevenDistribution, as it is equivalent to forceAllUnevenDistribution.

My experiment as mentioned above has shown otherwise. `forceUnevenDistribution` is true while `forceAllUnevenDistribution` is false when `Every option to receive a different number of points` is checked. Anyway, I will look into this to see why this is happening. 

I was just thinking instead of all this why don't we just have two variables (`forceAllUnevenDistribution` and `forceSomeUnevenDistribution`) corresponding to the two different constraints that could be checked. I won't modify anything right now but this is just a suggestion.

Reply:
>I was just thinking instead of all this why don't we just have two variables (forceAllUnevenDistribution and forceSomeUnevenDistribution) corresponding to the two different constraints that could be checked. I won't modify anything right now but this is just a suggestion.

I think you can do it this way, since if theres no restriction on input then theres no need to set any reminder for it.